improver/blending/weighted_blend.py

```diff
@@ -721,13 +718,28 @@ def weighted_mean(self, cube, weights):
         """
         weights_array = self.non_percentile_weights(cube, weights)
 
-        # Calculate the weighted average.
-        cube_new = collapsed(
-            cube, self.blend_coord, iris.analysis.MEAN, weights=weights_array
+        (collapse_dim,) = cube.coord_dims(self.blend_coord)
+        slice_dim = 1 if collapse_dim == 0 else 0
+        allow_slicing = cube.ndim > 1 and cube.shape[slice_dim] > 1
```

Comment:
Hmm, I'm having second thoughts on this condition. I think we want slicing to kick in only when certain size or number of dimensions is exceeded. Otherwise we might end up slicing over `y` for `x-y` cubes, which could add significant overhead (this may happen in threshold cli after #1259 refactor). How about we change that to:

```suggestion
        slicing_threshold = SOME_SUITABLE_NUMBER_HERE
        allow_slicing = cube.ndim > 2 and np.array(cube.shape).prod() > slicing_threshold
```